feat() add session based activity API, first steps for activity updates

stefandunca opened the pull request:
- [x] on top of https://github.com/status-im/status-go/pull/4658

This commit introduces the first steps towards implementing a session-based activity API to support dynamic updates of the current visualized filter in the wallet activity service. This change is necessary to move away from static paginated filtering, which was previously done in SQL, to a more dynamic approach that can handle updates in real-time.

The change is required by the new integration https://github.com/status-im/status-desktop/pull/13127 and doesn't affect the current API integration.

The main changes include:

- Added a `TODO.md` file outlining the plan and requirements for dynamic activity updates.
- New session-related API to the `activity.Service`
- `session.go` contains the logic for session management and event processing related to activity updates.
- Add test case for incremental filter updates.
    
The commit also includes:
    
- various other minor changes and refactoring to support the new session-based approach.
- Deprecation notices added to the `api.go` file for methods that are no longer used by the status-desktop application.
- Clarification comments added to the `scheduler.go` file regarding replacement policies.

Updates: [#12120](https://github.com/status-im/status-desktop/issues/12120)

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (78)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | f0f45144 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/1/) | 2023-12-17 04:18:15 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/1//consoleText) |
| :heavy_check_mark: | f0f45144 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/1/) | 2023-12-17 04:19:32 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231217-041541-f0f451-pr4480.zip) |
| :heavy_check_mark: | f0f45144 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/1/) | 2023-12-17 04:21:13 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231217-041536-f0f451-pr4480.aar) |
| :heavy_check_mark: | f0f45144 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/1/) | 2023-12-17 04:21:28 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231217-041536-f0f451-pr4480.zip) |
| | | | | | | |
| :heavy_multiplication_x: | f7876d00 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/2/) | 2023-12-17 04:25:39 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/2//consoleText) |
| :heavy_check_mark: | f7876d00 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/2/) | 2023-12-17 04:25:51 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231217-042316-f7876d-pr4480.aar) |
| :heavy_check_mark: | f7876d00 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/2/) | 2023-12-17 04:26:30 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231217-042320-f7876d-pr4480.zip) |
| :heavy_check_mark: | f7876d00 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/2/) | 2023-12-17 04:26:58 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231217-042320-f7876d-pr4480.zip) |
| | | | | | | |
| :heavy_check_mark: | 16eaf26d | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/3/) | 2023-12-20 02:18:30 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-021625-16eaf2-pr4480.zip) |
| :heavy_multiplication_x: | 16eaf26d | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/3/) | 2023-12-20 02:18:44 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/3//consoleText) |
| :heavy_check_mark: | 16eaf26d | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/3/) | 2023-12-20 02:21:47 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-021622-16eaf2-pr4480.aar) |
| | | | | | | |
| :heavy_multiplication_x: | d196033e | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/4/) | 2023-12-20 02:29:39 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/4//consoleText) |
| :heavy_check_mark: | d196033e | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/4/) | 2023-12-20 02:29:41 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231220-022815-d19603-pr4480.zip) |
| :heavy_check_mark: | d196033e | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/4/) | 2023-12-20 02:30:37 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231220-022815-d19603-pr4480.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 2053dd76 | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/5/) | 2023-12-27 19:01:37 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/5//consoleText) |
| :heavy_check_mark: | 2053dd76 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/5/) | 2023-12-27 19:02:15 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231227-185908-2053dd-pr4480.zip) |
| :heavy_check_mark: | 2053dd76 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/5/) | 2023-12-27 19:03:47 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231227-185903-2053dd-pr4480.aar) |
| :heavy_check_mark: | 2053dd76 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/5/) | 2023-12-27 19:04:11 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231227-185908-2053dd-pr4480.zip) |
| | | | | | | |
| :heavy_check_mark: | 518bfe35 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/6/) | 2023-12-27 19:08:29 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231227-190652-518bfe-pr4480.zip) |
| :heavy_multiplication_x: | 518bfe35 | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/6/) | 2023-12-27 19:09:14 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/6//consoleText) |
| :heavy_check_mark: | 518bfe35 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/6/) | 2023-12-27 19:10:30 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231227-190652-518bfe-pr4480.zip) |
| :heavy_check_mark: | 518bfe35 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/6/) | 2023-12-27 19:11:08 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231227-190652-518bfe-pr4480.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 510b429c | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/7/) | 2024-01-04 04:34:50 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/7//consoleText) |
| :heavy_check_mark: | 510b429c | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/7/) | 2024-01-04 04:35:45 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240104-043215-510b42-pr4480.zip) |
| :heavy_check_mark: | 510b429c | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/7/) | 2024-01-04 04:38:03 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240104-043211-510b42-pr4480.aar) |
| | | | | | | |
| :x: | ebbdf815 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/8/) | 2024-01-05 04:34:42 | ~15 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/8/consoleText) |
| :x: | ebbdf815 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/8/) | 2024-01-05 04:34:50 | ~24 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/8/consoleText) |
| :x: | ebbdf815 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/8/) | 2024-01-05 04:34:50 | ~23 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/8/consoleText) |
| :heavy_multiplication_x: | ebbdf815 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/8/) | 2024-01-05 04:36:30 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/8//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 76c919f8 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/9/) | 2024-01-05 17:53:48 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240105-175203-76c919-pr4480.zip) |
| :heavy_multiplication_x: | 76c919f8 | [#9](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/9/) | 2024-01-05 17:54:29 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/9//consoleText) |
| :heavy_check_mark: | 76c919f8 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/9/) | 2024-01-05 17:56:02 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240105-175203-76c919-pr4480.zip) |
| :heavy_check_mark: | 76c919f8 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/9/) | 2024-01-05 17:56:31 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240105-175158-76c919-pr4480.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 13cc1016 | [#10](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/10/) | 2024-01-06 04:10:01 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/10//consoleText) |
| :heavy_check_mark: | 13cc1016 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/10/) | 2024-01-06 04:12:13 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240106-040840-13cc10-pr4480.zip) |
| :heavy_check_mark: | 13cc1016 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/10/) | 2024-01-06 04:12:42 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240106-040840-13cc10-pr4480.zip) |
| :heavy_check_mark: | 13cc1016 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/10/) | 2024-01-06 04:13:24 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240106-040837-13cc10-pr4480.aar) |
| | | | | | | |
| :heavy_check_mark: | 5459de13 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/11/) | 2024-01-06 04:21:12 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240106-041951-5459de-pr4480.zip) |
| :heavy_multiplication_x: | 5459de13 | [#11](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/11/) | 2024-01-06 04:22:17 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/11//consoleText) |
| :heavy_check_mark: | 5459de13 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/11/) | 2024-01-06 04:22:18 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240106-041947-5459de-pr4480.aar) |
| :heavy_check_mark: | 5459de13 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/11/) | 2024-01-06 04:25:25 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240106-041947-5459de-pr4480.zip) |
| | | | | | | |
| :heavy_multiplication_x: | 5dcf453d | [#12](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/12/) | 2024-01-08 18:14:32 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/12//consoleText) |
| :heavy_check_mark: | 5dcf453d | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/12/) | 2024-01-08 18:15:53 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240108-181305-5dcf45-pr4480.aar) |
| :heavy_check_mark: | 5dcf453d | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/12/) | 2024-01-08 18:16:39 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240108-181310-5dcf45-pr4480.zip) |
| :heavy_check_mark: | 5dcf453d | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/12/) | 2024-01-08 18:20:28 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240108-181310-5dcf45-pr4480.zip) |
| | | | | | | |
| :heavy_multiplication_x: | f59fe7bc | [#13](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/13/) | 2024-01-08 18:33:23 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/13//consoleText) |
| :heavy_check_mark: | f59fe7bc | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/13/) | 2024-01-08 18:35:16 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240108-183155-f59fe7-pr4480.zip) |
| :heavy_check_mark: | f59fe7bc | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/13/) | 2024-01-08 18:35:42 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240108-183155-f59fe7-pr4480.zip) |
| :heavy_check_mark: | f59fe7bc | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/13/) | 2024-01-08 18:37:38 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240108-183155-f59fe7-pr4480.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 52a9e31e | [#14](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/14/) | 2024-01-08 21:26:15 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/14//consoleText) |
| :heavy_check_mark: | 52a9e31e | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/14/) | 2024-01-08 21:26:34 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240108-212454-52a9e3-pr4480.zip) |
| :heavy_check_mark: | 52a9e31e | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/14/) | 2024-01-08 21:28:28 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240108-212454-52a9e3-pr4480.zip) |
| :heavy_check_mark: | 52a9e31e | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/14/) | 2024-01-08 21:30:14 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240108-212454-52a9e3-pr4480.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 7af57063 | [#15](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/15/) | 2024-01-09 00:12:36 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/15//consoleText) |
| :heavy_multiplication_x: | 7af57063 | [#16](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/16/) | 2024-01-09 00:13:30 | ~47 sec | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/16//consoleText) |
| :heavy_check_mark: | 7af57063 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/15/) | 2024-01-09 00:13:58 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240109-001125-7af570-pr4480.aar) |
| :heavy_check_mark: | 7af57063 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/15/) | 2024-01-09 00:14:36 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240109-001125-7af570-pr4480.zip) |
| :heavy_check_mark: | 7af57063 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/15/) | 2024-01-09 00:14:49 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240109-001125-7af570-pr4480.zip) |
| :heavy_check_mark: | 7af57063 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/16/) | 2024-01-09 00:15:35 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240109-001401-7af570-pr4480.aar) |
| :heavy_check_mark: | 7af57063 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/16/) | 2024-01-09 00:16:17 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240109-001439-7af570-pr4480.zip) |
| :heavy_check_mark: | 7af57063 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/16/) | 2024-01-09 00:17:56 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240109-001452-7af570-pr4480.zip) |
| :heavy_check_mark: | 7af57063 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/17/) | 2024-01-09 00:23:25 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240109-002158-7af570-pr4480.zip) |
| :heavy_check_mark: | 7af57063 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/17/) | 2024-01-09 00:23:26 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240109-002155-7af570-pr4480.aar) |
| :heavy_multiplication_x: | 7af57063 | [#17](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/17/) | 2024-01-09 00:24:19 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/17//consoleText) |
| :heavy_check_mark: | 7af57063 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/17/) | 2024-01-09 00:25:00 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240109-002158-7af570-pr4480.zip) |
| :heavy_multiplication_x: | 7af57063 | [#18](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/18/) | 2024-01-10 18:44:32 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/18//consoleText) |
| :heavy_check_mark: | 7af57063 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/18/) | 2024-01-10 18:44:38 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240110-184308-7af570-pr4480.zip) |
| :heavy_check_mark: | 7af57063 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/18/) | 2024-01-10 18:46:20 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240110-184308-7af570-pr4480.zip) |
| :heavy_check_mark: | 7af57063 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/18/) | 2024-01-10 18:47:47 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240110-184308-7af570-pr4480.aar) |
| | | | | | | |
| :heavy_multiplication_x: | 9f1bd867 | [#19](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/19/) | 2024-01-13 00:43:06 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/19//consoleText) |
| :heavy_check_mark: | 9f1bd867 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/19/) | 2024-01-13 00:43:27 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240113-004148-9f1bd8-pr4480.zip) |
| :heavy_check_mark: | 9f1bd867 | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/19/) | 2024-01-13 00:44:10 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240113-004148-9f1bd8-pr4480.aar) |
| :heavy_check_mark: | 9f1bd867 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/19/) | 2024-01-13 00:47:21 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240113-004148-9f1bd8-pr4480.zip) |
| | | | | | | |
| :heavy_check_mark: | 388784bf | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/20/) | 2024-01-13 01:45:15 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240113-014304-388784-pr4480.aar) |
| :heavy_check_mark: | 388784bf | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/20/) | 2024-01-13 01:46:00 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240113-014308-388784-pr4480.zip) |
| :heavy_check_mark: | 388784bf | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/20/) | 2024-01-13 01:47:09 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240113-014308-388784-pr4480.zip) |
| :heavy_multiplication_x: | 388784bf | [#20](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/20/) | 2024-01-13 01:58:37 | ~15 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/20//consoleText) |
| :heavy_check_mark: | 388784bf | [#21](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/21/) | 2024-01-22 18:31:20 | ~39 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/21//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 314d769c | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/21/) | 2024-01-22 18:48:38 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240122-184529-314d76-pr4480.zip) |
| :heavy_check_mark: | 314d769c | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/21/) | 2024-01-22 18:49:04 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240122-184529-314d76-pr4480.zip) |
| :heavy_check_mark: | 314d769c | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/21/) | 2024-01-22 18:50:23 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240122-184529-314d76-pr4480.aar) |
| :heavy_check_mark: | 314d769c | [#22](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/22/) | 2024-01-22 19:22:54 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/22//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 7dcd9fb6 | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/22/) | 2024-02-01 04:40:56 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240201-043735-7dcd9f-pr4480.zip) |
| :heavy_check_mark: | 7dcd9fb6 | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/22/) | 2024-02-01 04:41:27 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240201-043735-7dcd9f-pr4480.zip) |
| :heavy_check_mark: | 7dcd9fb6 | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/22/) | 2024-02-01 04:42:13 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240201-043735-7dcd9f-pr4480.aar) |
| :heavy_check_mark: | 7dcd9fb6 | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/23/) | 2024-02-01 04:48:24 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240201-044525-7dcd9f-pr4480.zip) |
| :heavy_check_mark: | 7dcd9fb6 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/23/) | 2024-02-01 04:49:04 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240201-044525-7dcd9f-pr4480.zip) |
| :heavy_check_mark: | 7dcd9fb6 | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/23/) | 2024-02-01 04:50:16 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240201-044525-7dcd9f-pr4480.aar) |
| :heavy_check_mark: | 7dcd9fb6 | [#23](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/23/) | 2024-02-01 05:14:58 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/23//consoleText) |
| :heavy_check_mark: | 7dcd9fb6 | [#24](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/24/) | 2024-02-01 05:52:29 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/24//consoleText) |
| :heavy_check_mark: | 7dcd9fb6 | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4480/24/) | 2024-02-01 17:30:41 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-240201-172913-7dcd9f-pr4480.zip) |
| :heavy_check_mark: | 7dcd9fb6 | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4480/24/) | 2024-02-01 17:32:18 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-240201-172913-7dcd9f-pr4480.zip) |
| :heavy_check_mark: | 7dcd9fb6 | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4480/24/) | 2024-02-01 17:33:33 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-240201-172909-7dcd9f-pr4480.aar) |
| :heavy_multiplication_x: | 7dcd9fb6 | [#25](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/25/) | 2024-02-01 18:03:10 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/25//consoleText) |
| :heavy_check_mark: | 7dcd9fb6 | [#26](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/26/) | 2024-02-01 18:43:13 | ~36 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4480/26//consoleText) |

stefandunca committed:
feat(wallet): Add session-based activity API for dynamic updates

This commit introduces the first steps towards implementing a session-based activity API to support dynamic updates of the current visualized filter in the wallet activity service. This change is necessary to move away from static paginated filtering, which was previously done in SQL, to a more dynamic approach that can handle updates in real-time.

The main changes include:
- Add basic `EventActivitySessionUpdated` support for pending transactions.
- Added a `TODO.md` file outlining the plan and requirements for dynamic activity updates.
- New session-related API to the `activity.Service`
- `session.go` contains the logic for session management and event processing related to activity updates.
- Add test case for incremental filter updates.

The commit also includes:

- various other minor changes and refactoring to support the new session-based approach.
- Deprecation notices added to the `api.go` file for methods that are no longer used by the status-desktop application.
- Clarification comments added to the `scheduler.go` file regarding replacement policies.

Updates: #12120

ghstack-source-id: a61ef74184bbf826a748a3b8e6934a08a1c4bd86
Pull Request resolved: https://github.com/status-im/status-go/pull/4480

IvanBelyakoff commented on services/wallet/activity/service.go:
```diff
@@ -56,15 +62,35 @@ type Service struct {
 
-func NewService(db *sql.DB, tokenManager token.ManagerInterface, collectibles collectibles.ManagerInterface, eventFeed *event.Feed) *Service {
+func (s *Service) nextSessionID() SessionID {
+	s.lastSessionID.Add(1)
```
Add() returns a new value. `Load()` below is not needed

IvanBelyakoff commented on services/wallet/activity/service_test.go:
```diff
@@ -57,18 +58,25 @@ func (m *mockTokenManager) LookupToken(chainID *uint64, tokenSymbol string) (tkn
+
+	chainClient = transactions.NewMockChainClient()
+
+	pendingCheckInterval := time.Second
```
Why so long?

stefandunca replied:
Should be long enough that the second run is not executed given that it is not needed

IvanBelyakoff commented on services/wallet/activity/session.go:
```diff
@@ -0,0 +1,313 @@
+	Updated    []Entry         `json:"updated,omitempty"`
+}
+
+type fullFilter struct {
```
this type is exactly the same as Session type but model. Maybe make fullFilter an embedded type of Session?

stefandunca replied:
That could work. However, they are logically unrelated; `Session` is used to keep the session's state and `fullFilter` to provide a cleaner way to pass multiple parameters to `internalFilter` call. I renamed `fullFilter` to `fullFilterParams` to make it more verbose.

IvanBelyakoff commented on services/wallet/activity/session.go:
```diff
@@ -0,0 +1,313 @@
+		return a.Hex() < b.Hex()
+	})
+
+	s.sessionsRWMutex.Lock()
```
`func (s* Service) createSession` ?
The mutex locking will be handled better as well

stefandunca replied:
The implementation does more than just create the session. It also executes the SQL query and returns the first page.

IvanBelyakoff commented on services/wallet/activity/session.go:
```diff
@@ -0,0 +1,313 @@
+				})
+			}
+		}
+		s.sessionsRWMutex.Unlock()
```
deferred in the beginning of the func?

IvanBelyakoff commented on services/wallet/activity/session.go:
```diff
@@ -0,0 +1,313 @@
+
+// subscribeToEvents should be called with sessionsRWMutex locked for writing
+func (s *Service) subscribeToEvents() {
+	s.ch = make(chan walletevent.Event)
```
they say it should be buffered to avoid blocking other subscribers

stefandunca replied:
Indeed, thanks for the heads up.

IvanBelyakoff commented on services/wallet/activity/session.go:
```diff
@@ -0,0 +1,313 @@
+			}
+
+			for id := range s.sessions {
+				s.sessionsRWMutex.RLock()
```
I'd use as much as possible a C++ RAII-like technique of holding mutexes, for example put RLock/defer RUnlock call inside checkFilterForPending and Lock/defer Unlock inside addOnTop(). I see that it is not a member, well if it makes sense, you can create a member and move mutex locking there and call non-member from there.
That would be more clear imho and avoid duplicate Unlocking at different branches of if/else

stefandunca replied:
This will work but "hide" the locking at lower function call levels. I try not to spread the mutexes at all call levels and keep it as high as possible. I plan to go away from locking in the future PRs.

IvanBelyakoff commented on services/wallet/activity/session.go:
```diff
@@ -0,0 +1,313 @@
+	go s.processEvents()
+}
+
+func (s *Service) processEvents() {
```
Is this method meant to be looping forever until quit is called?

stefandunca replied:
Yes, it will loop until `event.Unsubscribe` is called which is done in `StopFilterSession`

IvanBelyakoff reviewed (commented):
Stefan, good job summing it up on what should be done and a plan for it!
I have a more technical review rather than a strategical, and there is a lot to think about as you mentioned it.
Please address the inline issues

stefandunca committed:
fix(wallet) prioritize type SendAT for pending entries in activity

In case both to/from addresses are present in the list we were using
the same logic as for transfers. However, this doesn't make sense given
that we can have only one entry in pending activity.

The following cases are still covered

- When the receiver is in addresses we get received
- When both receiver and sender are in the list will get sent
- When the sender is on the list we will get sent

Updates status-desktop #12120

stefandunca committed:
fix(wallet) fix reading amount for pending transactions

The reading of the amount for pending transactions was done in the same
way as for transfers table. However, the transfers table has a string
hex representation of the amount, while the pending transactions table
has a binary representation of the amount (*big.Int). This was
triggering the not int warning and value was missing.

Updates status-desktop #12120
